File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/other_settings/checkstyle/CheckstyleSettings.java
package com.github.alexandrecarlton.idea.settings.layout.other_settings.checkstyle;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.List;
import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableCheckstyleSettings.class)
public interface CheckstyleSettings {

  Optional<String> checkstyleVersion();

  Optional<CheckstyleScanScope> scanScope();

  Optional<Boolean> treatCheckstyleErrorsAsWarnings();

  List<CheckstyleConfigurationFile> configurationFiles();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/other_settings/checkstyle/CheckstyleSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.other_settings.checkstyle

import com.fasterxml.jackson.annotation.JsonProperty

data class CheckstyleSettings(

    @JsonProperty("Checkstyle version")
    val checkstyleVersion: String? = null,

    @JsonProperty("Scan Scope")
    val scanScope: CheckstyleScanScope? = null,

    @JsonProperty("Treat Checkstyle errors as warnings")
    val treatCheckstyleErrorsAsWarnings: Boolean? = null,

    @JsonProperty("Configuration files")
    val configurationFiles: List<CheckstyleConfigurationFile>? = null
)
